### PR TITLE
`_collect_linear()` does not account for fixed variables' coefficients

### LINKED ISSUE #1399: `_collect_linear()` does not account for fixed variables' coefficients
`_collect_linear()` is not computing the `constant` term correctly when fixed variables have coefficients. This is resulting in the wrong LP file being written when using `quicksum()` (or `LinearExpression`s in general) in constraint rules.

### Current

```python
for c,v in zip(exp.linear_coefs, exp.linear_vars):
    if v.fixed:
        if compute_values:
            ans.constant += multiplier*v.value
        else:
            ans.constant += multiplier*v
    ...
```

### Suggested

```python
for c,v in zip(exp.linear_coefs, exp.linear_vars):
    if v.fixed:
        if compute_values:
            ans.constant += multiplier * c * value(v)
        else:
            ans.constant += multiplier * c * v
    ...
```

### PR DESCRIPTION
**Pull request by ruaridhw**
## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/repn/tests/test_standard.py [hunk 1]
```diff
@@ -4010,9 +4010,9 @@ def test_linear1(self):
 
         m.v[1].fixed=True
         rep = generate_standard_repn(e, compute_values=True)
-        self.assertEqual(str(rep.to_expression()), "2 + v[3] + 2*v[4] + 3*v[5]")
+        self.assertEqual(str(rep.to_expression()), "-1 + v[3] + 2*v[4] + 3*v[5]")
         rep = generate_standard_repn(e, compute_values=False)
-        self.assertEqual(str(rep.to_expression()), "v[1] + v[1] + v[3] + 2*v[4] + 3*v[5]")
+        self.assertEqual(str(rep.to_expression()), "v[1] - 2*v[1] + v[3] + 2*v[4] + 3*v[5]")
```
Comment (ruaridhw) [inline pyomo/repn/tests/test_standard.py:4015]: Unless I'm mistaken, this test was asserting that the behaviour was wrong.

If the expression is `- v[1] + v[3] + 2*v[4] + 3*v[5]` and we fix `v[1]` to `1` then I can't see why the constant term is `2`.

Comment (michaelbynum) [reply pyomo/repn/tests/test_standard.py:4015]: @ruaridhw I agree.

File: pyomo/repn/standard_repn.py [hunk 2]
```diff
@@ -867,9 +867,9 @@ def _collect_linear(exp, multiplier, idMap, compute_values, verbose, quadratic):
     for c,v in zip(exp.linear_coefs, exp.linear_vars):
         if v.fixed:
             if compute_values:
-                ans.constant += multiplier*v.value
+                ans.constant += multiplier * c * value(v)
```
Comment (michaelbynum) [inline pyomo/repn/standard_repn.py:870]: @ruaridhw If `compute_values` is True, then we probably want `value(c)` in case c is a parameter.

Comment (ruaridhw) [reply pyomo/repn/standard_repn.py]: Good point, thanks!

**Pull request merged**